GGRC-6213 Reviewers should be in plural on UI

SaratPavan417 opened the pull request:
<!-- If your PR depends on other tickets or PRs, you can list them here
# Dependencies

This PR is `on hold` until the dependencies are merged:

- [ ] GGRC-1234
- [ ] #1234
-->

# Issue description

"Reviewer' field is displayed on "Request Review" pop-up which should be re-named as 'Reviewers' in all places on UI for all objects that have review.

# Steps to test the changes

Go to My Work page
Click "Controls" tab.
Click on a control raw.
Click "Request Review" button.
- "Reviewer' field is displayed on "Request Review" pop-up.

# Solution description

Update the file "/vagrant/src/ggrc/migrations/versions/20190306113217_adf7bdb8996e_rename_reviewer_to_reviewers.py" with the below sql query:
Query : UPDATE access_control_roles SET name = 'REVIEWERS' WHERE parent_id is NULL and name = "REVIEWER";

Note: If the file is not editable change the permissions by logging out of the docker container.

Run the below command in order to execute the sql changes:
python src/ggrc/migrate.py revision -m "Rename reviewer to reviewers" 

Run the below query to verify that the changes were made successfully:
select * from access_control_roles where parent_id is NULL and name = "REVIEWERS";

Once the changes are verified, run the command:
 - db_reset

# Sanity checklist

- [x] I have clicked through the app to make sure my changes work and not break the app.
- [ ] I have applied the correct milestone and labels.
- [x] My changes fix the issue described in the description (and do nothing else). 🤞
- [x] My changes are covered by tests.
- [ ] My changes follow our [performance guidelines](https://github.com/google/ggrc-core/blob/dev/docs/source/contributing/performance.rst).
- [ ] My changes follow our [js](https://github.com/google/ggrc-core/blob/dev/docs/source/contributing/javascript.rst) and/or [python](https://github.com/google/ggrc-core/blob/dev/docs/source/contributing/python.rst) guidelines.
- [ ] My commits follow our [commit guidelines](https://github.com/google/ggrc-core/blob/dev/docs/source/contributing/git/how_to_write_a_commit_message.rst).

<!-- If your PR includes a migration include the additional checklist items
# Migration checklist
- [ ] Migration passes all checks from our [PR review guidelines](https://github.com/google/ggrc-core/blob/dev/docs/source/contributing/git/reviewing_pull_requests.rst#reviewing-a-pr-containing-database-migration-scripts)
- [ ] Upon merging, add 'check migration chain' and 'kokoro:force-run' labels to all open PRs with a label 'migration'
- [ ] Upon merging, update 'table structure changes' in weekly deployment documentation
-->

# PR Review checklist

- [x] The changes fix the issue and don't cause any apparent regressions.
- [x] Labels and milestone are correctly set.
- [x] The solution description matches the changes in the code.
- [x] There is no apparent way to improve the performance & design of the new code.
- [x] The pull request is opened against the correct base branch.
- [x] Upon merging, the Jira ticket's fixversion is correctly set and the ticket is moved to "QA - In Progress".

<!-- If your code is not finished yet can include a TODO check list
# TODO

- [ ] First item on the TODO
-->

SaratPavan417 commented:
I signed it!

NastassiaKiseleva commented:
@SaratPavan417 @zidarsk8 Please verify that search/filtering, import export also work well together with this change.

zidarsk8 commented on src/ggrc/migrations/versions/20190306113217_adf7bdb8996e_rename_reviewer_to_reviewers.py:
```diff
@@ -0,0 +1,30 @@
+def upgrade():
+  """Upgrade database schema and/or data, creating a new revision."""
+  conn = op.get_bind()
+  res = conn.execute("UPDATE access_control_roles SET name = 'REVIEWERS' WHERE parent_id is NULL and name = 'REVIEWER'")
```
I am guessing you took the caps from the frontend, but in different places, we might display this role differently.

So what should have been done is check how the role is currently stored and then modify that to fit the new requirement. And here we can see that the word is capitalized and not in all caps.

```
mysql> select id, name from access_control_roles where name = "reviewer";
+------+----------+
| id   | name     |
+------+----------+
| 4160 | Reviewer |
+------+----------+
1 row in set (0.00 sec)
```

PS: one place where this role would look wrong would be on the admin page. For now we don't have custom roles on reviews but if that changes we'd have a bug with a weird role name.

belovmd commented:
@zidarsk8 shouldn't we also create revisions for affected records and implement logic for revisions content populate?

zidarsk8 commented:
@belovmd Revisions don't store role names and those aren't used, they only store role ids and names are pulled from the db even for history log.

zidarsk8 commented:
@SaratPavan417 can you please rebase your changes on top of the latest develop branch and update migration chain (down revision needs to be update to the latest one)

SaratPavan417 commented:
> @SaratPavan417 can you please rebase your changes on top of the latest develop branch and update migration chain (down revision needs to be update to the latest one)

@zidarsk8 Updated the down revision to the latest. Rebased my changes on top of the latest develop branch and updated the migration chain

victoria-kostyukova commented:
@SergeyRabtsau please review https://github.com/victoria-kostyukova/ggrc-core/commit/626697b34b388ee104af9e85a51b46d28359c3b7

SergeyRabtsau commented:
@victoria-kostyukova  :+1: Selenium part LGTM :tada:
@SaratPavan417 pls add commit to your PR.

zidarsk8 commented:
Adding needs work, to make the git commits nicer.
